### PR TITLE
awscli: remove groff dep (and hint at removing for awscli@1)

### PR DESCRIPTION
**Pull request by razamatan**
- 2.8.11 supports mandoc (https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst#L11)
- 1.27.6 supports mandoc (https://github.com/aws/aws-cli/blob/develop/CHANGELOG.rst#L19)
  so, just comment to remove it later (couldn't figure out how to version check

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/awscli.rb [hunk 1]
```diff
@@ -22,10 +22,6 @@ class Awscli < Formula
   depends_on "python@3.10"
   depends_on "six"
 
-  on_system :linux, macos: :ventura_or_newer do
-    depends_on "groff"
-  end
-
```
Comment (Bo98) [inline Formula/awscli.rb:28]: Replace this with `uses_from_macos "mandoc"`.

Comment (razamatan) [reply Formula/awscli.rb]: thank you!  i was trying to figure out how to get the ci to pass on ubuntu...

**Comment by razamatan** [comment]
this needs https://github.com/Homebrew/brew/pull/14137

**Comment by razamatan** [comment]
ptal

**Review by Bo98 — APPROVED** [review]
Thanks!

**Pull request closed without merging**